Tasks/SonarQubePostTest/SonarQubeMetrics.ps1

```diff
@@ -38,18 +38,38 @@ function WaitForAnalysisToFinish
     Assert (![String]::IsNullOrEmpty($analysisId)) "Could not fetch the analysis id"
 }
 
+
+function FetchQualityGateDetails
+{
+    param ([Array]$errors,[Array]$warnings)
+    
+    $analysisId = GetTaskContextVariable "MSBuild.SonarQube.AnalysisId"             
+    Assert (![String]::IsNullOrEmpty($analysisId)) "WaitForAnalysisToFinish should be called first."
```

Comment:
If you're fetching id in multiple places then it might be worth putting lines 46 and 47 into a separate function.